pkg/registry/apis/secret/secretkeeper/secretkeeper.go

```diff
@@ -3,43 +3,37 @@ package secretkeeper
 import (
 	"fmt"
 
+	secretv0alpha1 "github.com/grafana/grafana/pkg/apis/secret/v0alpha1"
 	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/grafana/grafana/pkg/registry/apis/secret/contracts"
 	"github.com/grafana/grafana/pkg/registry/apis/secret/secretkeeper/sqlkeeper"
 )
 
-// Service is the interface for secret keeper services.
-// This exists because OSS and Enterprise have different amounts of keepers available.
-type Service interface {
-	GetKeepers() (map[contracts.KeeperType]contracts.Keeper, error)
-}
-
 // OSSKeeperService is the OSS implementation of the Service interface.
 type OSSKeeperService struct {
-	tracer            tracing.Tracer
-	encryptionManager contracts.EncryptionManager
-	store             contracts.EncryptedValueStorage
+	systemKeeper *sqlkeeper.SQLKeeper
 }
 
+var _ contracts.KeeperService = (*OSSKeeperService)(nil)
+
 func ProvideService(
 	tracer tracing.Tracer,
 	store contracts.EncryptedValueStorage,
 	encryptionManager contracts.EncryptionManager,
-) (OSSKeeperService, error) {
-	return OSSKeeperService{
-		tracer:            tracer,
-		encryptionManager: encryptionManager,
-		store:             store,
-	}, nil
-}
-
-func (ks OSSKeeperService) GetKeepers() (map[contracts.KeeperType]contracts.Keeper, error) {
-	sqlKeeper, err := sqlkeeper.NewSQLKeeper(ks.tracer, ks.encryptionManager, ks.store)
+) (*OSSKeeperService, error) {
+	// TODO: rename to system keeper or something like that
+	systemKeeper, err := sqlkeeper.NewSQLKeeper(tracer, encryptionManager, store)
```

Comment:
Claim: we can instantiate the SQL storage here when providing the service right? I don't  see why we MUST do it when calling `GetKeepers()`.